conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt

```diff
@@ -0,0 +1,1330 @@
+Labels: binary-bitwise-expr, byte, DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, var
+
+function init() {
+    int a = 9876;
+    int:Unsigned32 b = 5454;
+    var c = a & b;
+    int:Unsigned32 result1 = c;
+    io:println(result1); // @output 1028
+
+    int:Unsigned16 d = 4344;
+    var e = b & d;
+    int:Unsigned16 result2 = e;
+    io:println(result2); // @output 4168
+
+    int:Unsigned8 f = 44;
+    var g = d & f;
+    int:Unsigned8 result3 = g;
+    io:println(result3); // @output 40
+
+    var h = f & g;
+    int:Unsigned8 result4 = h;
+    io:println(result4); // @output 40
+
+    var i = f & g;
+    byte result5 = h;
+    io:println(result5); // @output 40
+}
+
+Test-Case: error
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K using invalid assignment.
+Labels: binary-bitwise-expr, byte, DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, 
+        int:Signed32
+
+function init() {
+    int a = 4343;
+    int b = 8;
+    int:Unsigned32 _ = a & b; // @error the static type of the result is int
+
+    int d = 9876;
+    int:Unsigned32 e = 5454;
+    int:Unsigned16 _ = d & e; // @error smallest K of the operands is 32, hence the static type of the result is Unsigned32
+
+    int g = 9876;
+    int:Unsigned16 h = 5454;
+    int:Unsigned8 _ = g & h; // @error smallest K of the operands is 16, hence the static type of the result is Unsigned16
+
+    byte _ = g & h; // @error smallest K of the operands is 16, hence the static type of the result is Unsigned16
+
+    int:Signed8 k = 12;
+    int:Signed16 m = 4434;
+    int:Signed8 _ = k & m; // @error the static type of the result is int
+
+    int:Unsigned16 n = 12;
+    int:Unsigned16 p = 4434;
+    int:Signed16 _ = n & p; // @error the static type of the result is Unsigned16
+
+    int:Unsigned32 q = 12;
+    int:Unsigned16 r = 4434;
+    int:Unsigned8 _ = q & r; // @error the static type of the result is Unsigned16
+}
+
+Test-Case: error
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K via invalid assignment with inference.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, 
+        int:Signed16, var
+
+function init() {
+    int a = 4343;
+    int b = 8;
+    var result1 = a & b;
+    int:Unsigned32 _ = result1; // @error the static type of the result is int
+
+    int d = 9876;
+    int:Unsigned32 e = 5454;
+    var result2 = d & e;
+    int:Unsigned16 _ = result2; // @error smallest K of the operands is 32
+
+    int g = 9876;
+    int:Unsigned16 h = 5454;
+    var result3 = g & h;
+    int:Unsigned8 _ = result3; // @error smallest K of the operands is 16
```

Comment:
The idea for error tests is to be able to compare this text and the actual error message and validate that it is indeed the correct error message. Shall we rethink these errors at least similar to L503?